### ISSUE TITLE
Implement new Concept Exercise: Dates

### ISSUE THREAD

**Issue by junedev** [exercism/javascript#1528]
Note: The concept was already created, only the exercise is missing now.

## Getting Started

If you have not yet contributed to concept exercises before, this task requires some upfront reading to acquire the necessary background knowledge.

Here you can read about what Concept Exercises are and how they are structured:

- [What are Concept Exercises?](https://exercism.org/docs/building/product/concept-exercises)
- [Concept Exercises](https://exercism.org/docs/building/tracks/concept-exercises)
- [Concepts](https://exercism.org/docs/building/tracks/concepts)

If you have not done so yet, it is probably also helpful to do a couple of "Learning Exercises" (this is how they are called on the site) yourself. You can also look at the code of an existing concept exercise like [`bird-watcher`](https://github.com/exercism/javascript/tree/main/exercises/concept/bird-watcher) (concept [`for-loops`](https://github.com/exercism/javascript/tree/main/concepts/for-loops)) for reference.

See the documentation _above_ (general documentation), as well as [How to implement a Concept Exercise in JavaScript](https://github.com/exercism/javascript/blob/main/reference/implementing-a-concept-exercise.md).

Also be aware of these general guidelines.

- [Exercism Markdown Specification](https://exercism.org/docs/building/markdown/markdown)
- [Exercism Formatting and Style Guide](https://exercism.org/docs/building/markdown/style-guide)

## Goal

The goal here is to create a new concept exercise that teaches working with dates via `Date`.

The concept needs to be written from scratch, the exercise can be ported from another track (see details below).

## Concepts

The following concept needs to be created. You can use the `introduction.md` file of the concept also as `introduction.md` file of the exercise. No need to create different content at this point. Additionally, if you want to save some time it is ok to not have an extensive `about.md` for now. It can also be mainly the `introduction.md` content, maybe with some additions you would like to make.

- `dates` (maybe use "Date and Time" as display name in the config)

## Learning Objectives

In the concepts the student should learn about the following topics and then practice them in the concept exercise.

- how dates/timestamps are represented internally, see e.g. [MDN link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_ecmascript_epoch_and_timestamps)
- how to create dates with `new Date`, incl. some of the commonly used [variants](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax)
- `Date.parse`
- `Date.now`
- getting and setting the date components (`getMonth`/`setMonth` etc.)
- how to calculate a [time difference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#calculating_elapsed_time)
- how to compare dates

The about.md file could additionally mention ...

- how out of range inputs are handled, see https://javascript.info/date#autocorrection
- [Temporal](https://tc39.es/proposal-temporal/docs/index.html) as work-in-progress to address shortcoming of working with dates in JavaScript

## Out of Scope

- Introduction to [Temporal](https://tc39.es/proposal-temporal/docs/index.html)
- [Internationalization API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat)
- Third party packages for handling dates

## Prerequisites

- `classes`

Other prerequisites should be added as needed to solve the specific exercise.

## Exercise Idea

There are two options for exercises from other languages that could be used as a template.

- [Elixir "Library Fee" Exercise](https://github.com/exercism/elixir/tree/main/exercises/concept/library-fees)
- [C# "Booking up for Beauty" Exercise](https://github.com/exercism/csharp/tree/main/exercises/concept/booking-up-for-beauty)

If you port one of these exercises, make sure only to include tasks that actually make sense in JavaScript and add value for the student. Feel free to remove/replace/add tasks as needed.

## Resources

Here some links that might be helpful as a starting point and/or for the links section of the concept:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
- https://javascript.info/date

## How to proceed

1. **First accept this issue by saying "I'd like to work on this"** (no need to wait for a response, just go ahead).
2. Use this issue to discuss any questions you have, what should be included in the content and what not and to collect more reference material.
3. Create a PR and set "exercism/javascript" as reviewers. Additionally you can write in #maintaining-javascript on Slack that your PR is ready for review. Once you incorporated any critical feedback that the reviewer might give you and the PR is approved, it will be merged by a maintainer.

**Comment by JaPatGitHub** [comment]
I'd like to work on this!!

**Comment by JaPatGitHub** [comment]
Hey @junedev! Just wanted to ask if we can merge the concept first and then exercise?
Or both together?

**Comment by junedev** [comment]
Yes, you can create a separate PR for the concept first if you like.

**Comment by JaPatGitHub** [comment]
Hey @junedev, @SleeplessByte, an issue here
I ran the command `npx babel-node scripts/format` from my Windows PowerShell terminal. It gives this output

```
npx: installed 1 in 12.205s
command not found: babel-node
```

What to do?

**Comment by SleeplessByte** [comment]
I think it's because there is no PS1 variant in the scripts folder.

Do you happen to have git bash? That's what I use on my windows to run these. Alternatively, it should work in CMD.exe.

If all else fails:

npm install
prettier --write .

Or add a comment to the PR with a slash followed by `format`.

**Comment by JaPatGitHub** [comment]
Hey @junedev ! The display on my laptop doesn't work and would take few days to fix. Hence temporarily won't be able to work on this issue and the linked PR.
Hope that doesn't cause any trouble?

I'll work on it asap

**Comment by junedev** [comment]
No problem. Good luck with the repair/replacement.

**Comment by junedev** [comment]
@JaPatGitHub Now that the concept part is merged, please let me know whether you intend to also work on the exercise.

**Comment by JaPatGitHub** [comment]
Yes @junedev , I am ready work on the exercises too :)

**Comment by junedev** [comment]
Great! Then I leave the assignment as is.

**Comment by JaPatGitHub** [comment]
@junedev @SleeplessByte, I am unable to find time to work on this further owing to my heavy courseload as this is the final year of my high-school. 

I apologize for the inconvenience caused and for informing this late...I try to scrape out time to just complement this assignment but fail to make sufficient progress.

**Comment by junedev** [comment]
👍 No problem, I will unassign you.

**Comment by SalahuddinAhammed** [comment]
I'd like to work on this.

**Comment by SleeplessByte** [comment]
This was done.
